crm_opportunity_currency/models/crm_lead.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    customer_currency_id = fields.Many2one(
+        string='Customer currency',
+        comodel_name='res.currency',
+        default=lambda self: self.env.user.company_id.currency_id,
```

Comment:
Maybe add `ondelete='restrict`, what do you think? (the default value is `set null`)

Reply:
@sebalix yes (but in this specific case, chances are really high that some other required currency_id field on another model will prevent deleting the currency record, which is not something you do very often :smile_cat: )